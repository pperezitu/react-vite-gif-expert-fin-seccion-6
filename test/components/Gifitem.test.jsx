import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";


describe('Pruebas de <Giftitem />', () => {

    const title = 'Saitama es la cumbia';
    const url = 'https://one-puch.com/saitama.png';
    const giftItem = render(<GifItem title={title} url={url} />);

    test('debe hacer match con el snapshot', () => {
        const { container } = giftItem;
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar el URL y el ALT indicado', () => {
        render(<GifItem title={title} url={url} />);
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    });

    test('debe mostrar el titulo del componente', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });

});