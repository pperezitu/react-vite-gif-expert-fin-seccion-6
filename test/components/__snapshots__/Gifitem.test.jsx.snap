// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <Giftitem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama es la cumbia"
      src="https://one-puch.com/saitama.png"
    />
    <p>
      Saitama es la cumbia
    </p>
  </div>
</div>
`;
